### PR TITLE
Remove old str.format() syntax and change to f-Strings

### PR DESCRIPTION
**Pull request by PhilippImhof**
## Motivation

I just came across a few strings that were still written in the old-fashioned way.

## Overview / Explanation for Changes

Removed (almost) all old-fashioned uses of `str.format(...)`, except for cases where there is no easy f-String solution, e.g. in combination with `**` operator (e.g. `path.format(**all_args)`) or where the definition of the string is done at a completely different place like with `super().format(record_c)` or `constants.SCENE_NOT_FOUND_MESSAGE.format(scene_name)`

## Testing Status

Tested locally, including doctests.

## Acknowledgements
- [X] I have read the [Contributing Guidelines](https://docs.manim.community/en/latest/contributing.html)

## Reviewer Checklist
- [x] Newly added functions/classes are either private or have a docstring
- [x] Newly added functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [x] Newly added documentation builds, looks correctly formatted, and adds no additional build warnings
- [ ] The oneline summary has been included [in the wiki](https://github.com/ManimCommunity/manim/wiki/Changelog-for-next-release)

### CODE REVIEW

File: manim/mobject/svg/code_mobject.py [hunk 1]
```diff
@@ -263,9 +263,7 @@ def ensure_valid_file(self):
             if os.path.exists(path):
                 self.file_path = path
                 return
-        error = "From: {}, could not find {} at either of these locations: {}".format(
-            os.getcwd(), self.file_name, possible_paths
-        )
+        error = f"From: {os.getcwd()}, could not find {self.file_name} at either of these locations: {possible_paths}"
```
Comment (behackl) [inline manim/mobject/svg/code_mobject.py:266]: This line seems a little long, but if black doesn't complain it should still be < 90 chars. (Could you please check, and if it is longer, break it up?)

**Review by behackl — APPROVED** [review]
LGTM, just found one tiny thing to (potentially) complain about. :-)

File: tests/utils/GraphicalUnitTester.py [hunk 2]
```diff
@@ -76,7 +76,7 @@ def _load_data(self):
             The pre-rendered frame.
         """
         frame_data_path = os.path.join(
-            os.path.join(self.path_control_data, "{}.npz".format(str(self.scene)))
+            os.path.join(self.path_control_data, f"{str(self.scene)}.npz")
```
Comment (kolibril13) [inline tests/utils/GraphicalUnitTester.py:79]: This could also be changed to pathlib, but this can also happen in a future pr :)

**Comment by behackl** [comment]
Oh, also: if you don't provide a separate one-line summary, I'll simply use the PR title for the corresponding changelog entry.

File: manim/mobject/mobject.py [hunk 3]
```diff
@@ -1254,9 +1254,10 @@ def construct(self):
     # Errors
     def throw_error_if_no_points(self):
         if self.has_no_points():
-            message = "Cannot call Mobject.{} " + "for a Mobject with no points"
             caller_name = sys._getframe(1).f_code.co_name
-            raise Exception(message.format(caller_name))
+            raise Exception(
+                f"Cannot call Mobject.{caller_name} for a Mobject with no points"
+            )
```
Comment (kolibril13) [inline manim/mobject/mobject.py:1260]: so much better!

File: manim/scene/scene_file_writer.py [hunk 4]
```diff
@@ -399,7 +396,7 @@ def is_already_cached(self, hash_invocation):
             return False
         path = os.path.join(
             self.partial_movie_directory,
-            "{}{}".format(hash_invocation, config["movie_file_extension"]),
+            f"{hash_invocation}{config['movie_file_extension']}",
```
Comment (kolibril13) [inline manim/scene/scene_file_writer.py:399]: Is there a reason for changing " into ' for movie_file_extension?

File: manim/utils/tex_file_writing.py [hunk 5]
```diff
@@ -126,8 +126,8 @@ def tex_compilation_command(tex_compiler, output_format, tex_file, tex_dir):
             outflag,
             "-interaction=batchmode",
             "-halt-on-error",
-            '-output-directory="{}"'.format(tex_dir),
-            '"{}"'.format(tex_file),
+            f'-output-directory="{tex_dir}"',
+            f'"{tex_file}"',
```
Comment (kolibril13) [inline manim/utils/tex_file_writing.py:130]: Why is here the order  `'"{tex_file}"'` and not  `"'{tex_file}'"`

File: manim/utils/tex_file_writing.py [hunk 6]
```diff
@@ -197,7 +197,7 @@ def convert_to_svg(dvi_file, extension, page=1):
             "dvisvgm",
             "--pdf" if extension == ".pdf" else "",
             "-p " + str(page),
-            '"{}"'.format(dvi_file),
+            f'"{dvi_file}"',
```
Comment (kolibril13) [inline manim/utils/tex_file_writing.py:200]: same comment as above : why '"x"'?

**Review by kolibril13 — APPROVED** [review]
Some small comments, but all in all a nice addition!

File: manim/scene/scene_file_writer.py (hunk 4 above)
Comment (PhilippImhof) [reply manim/scene/scene_file_writer.py:399]: No particular reason. The string was delimited by double quotes, so I used the single quotes for the inner part.

File: manim/utils/tex_file_writing.py (hunk 5 above)
Comment (PhilippImhof) [reply manim/utils/tex_file_writing.py:130]: Same as above: I wanted to stick to the format used before. Probably not a good idea in hindsight.

File: manim/utils/tex_file_writing.py (hunk 6 above)
Comment (PhilippImhof) [reply manim/utils/tex_file_writing.py:200]: The file name will be set between double quotes for the shell, so I think it makes sense to have the outer quotes being single quotes

File: manim/mobject/svg/code_mobject.py (hunk 1 above)
Comment (PhilippImhof) [reply manim/mobject/svg/code_mobject.py:266]: Funny... Black did not change it (neither locally nor in the repository), but the line was clearly longer than 90 chars.

File: tests/utils/GraphicalUnitTester.py (hunk 2 above)
Comment (PhilippImhof) [reply tests/utils/GraphicalUnitTester.py:79]: Good point. I would prefer to not mix those two things.

File: manim/utils/tex_file_writing.py (hunk 5 above)
Comment (PhilippImhof) [reply manim/utils/tex_file_writing.py:130]: > Oh, also: if you don't provide a separate one-line summary, I'll simply use the PR title for the corresponding changelog entry.

Actually, I did not consider the change significant enough to be worth an entry to the changelog....

**Comment by PhilippImhof** [comment]
> Oh, also: if you don't provide a separate one-line summary, I'll simply use the PR title for the corresponding changelog entry.

Actually, I did not consider the change significant enough to be included in the changelog...

**Comment by behackl** [comment]
> > Oh, also: if you don't provide a separate one-line summary, I'll simply use the PR title for the corresponding changelog entry.
> 
> Actually, I did not consider the change significant enough to be included in the changelog...

I don't want to (have to) make the decision, when a PR is significant enough in order to be included in the changelog, which is why for now I am including all merged PRs. (I also think it is nice to have a complete list of PRs that have been merged before a release. :-))

**Comment by PhilippImhof** [comment]
Modifications after the merge of @eulertour's  #830. I think it would be good if someone could double check, just to make sure I did not introduce some errors while solving the conflicts. I tried my best not to, of course.

File: tests/utils/GraphicalUnitTester.py (hunk 2 above)
Comment (behackl) [inline tests/utils/GraphicalUnitTester.py:79]: The call to `str` within the f-string can be removed:

```suggestion
            os.path.join(self.path_control_data, f"{self.scene}.npz")
```

**Review by behackl — APPROVED** [review]
Still LGTM.

File: manim/scene/scene_file_writer.py (hunk 4 above)
Comment (kolibril13) [reply manim/scene/scene_file_writer.py:399]: okay!

File: manim/utils/tex_file_writing.py (hunk 6 above)
Comment (kolibril13) [reply manim/utils/tex_file_writing.py:200]: alright. I just found a source that this is possible to do it like this in python:
If either single or double quote is a part of the string itself, then the string must be placed in double or single quotes respectively.
https://www.tutorialspoint.com/What-is-the-difference-between-single-and-double-quotes-in-python#:~:text=What%20is%20the%20difference%20between%20single%20and%20double%20quotes%20in%20python%3F,-PythonServer%20Side&text=In%20Python%2C%20such%20sequence%20of,representations%20can%20be%20used%20interchangeably.

File: manim/utils/tex_file_writing.py (hunk 5 above)
Comment (kolibril13) [reply manim/utils/tex_file_writing.py:130]: Alright!

File: tests/utils/GraphicalUnitTester.py (hunk 2 above)
Comment (PhilippImhof) [reply tests/utils/GraphicalUnitTester.py:79]: Good catch!

**Pull request merged**